lib/absinthe/phase/error.ex

```diff
@@ -15,18 +16,21 @@ defmodule Absinthe.Phase.Error do
     message: String.t,
     phase: module,
     locations: [loc_t],
+    extra: Keyword.t
   }
 
   @doc """
   Generate a phase error that relates to a specific point in the
   document.
   """
   @spec new(Absinthe.Phase.t, String.t, loc_t | [loc_t]) :: t
-  def new(phase, message, location) do
+  @spec new(Absinthe.Phase.t, String.t, loc_t | [loc_t], Keyword.t) :: t
+  def new(phase, message, location, extra \\ []) do
     %__MODULE__{
       phase: phase,
       message: message,
-      locations: List.wrap(location)
+      locations: List.wrap(location),
+      extra: Enum.filter(extra, &filter_extra_item/1)
```

Comment:
I don't think filtering to enforce JSON compatibility makes sense here.

It's quite possible for people to use Absinthe w/out serializing to JSON. Absinthe doesn't even _know_ about JSON serialization -- that's all handled by Absinthe.Plug.

Personally, I'd opt for things blowing up instead of silent removal of information, which doesn't feel very idiomatic -- but if filtering is important, I think it should be a separate, general purpose phase (focused on more than just errors, probably) that's packaged with Absinthe.Plug. Thoughts, @benwilson512 ?